core/modules/bypass.cc

```diff
@@ -30,7 +30,37 @@
 
 #include "bypass.h"
 
+CommandResponse Bypass::Init(const bess::pb::BypassArg &arg) {
+  cycles_per_batch_ = arg.cycles_per_batch();
+  cycles_per_packet_ = arg.cycles_per_packet();
+
+  cycles_per_byte_ = arg.cycles_per_byte();
+  if (cycles_per_byte_ < 0) {
+    return CommandFailure(EINVAL, "Cycles must be equals to/greater than zero");
+  }
+
+  return CommandSuccess();
+}
+
 void Bypass::ProcessBatch(bess::PacketBatch *batch) {
+  uint32_t cur_tsc = ctx.current_tsc();
+  uint32_t cycles = cycles_per_batch_ + cycles_per_packet_ * batch->cnt();
+
+  if (cycles_per_byte_) {
+    uint64_t total_bytes = 0;
+    for (int i = 0; i < batch->cnt(); i++) {
```

Comment:
Shouldn't we subtract out the cycles spent doing this calculation? The extra call to `rdtsc()` might not make sense for low values of `cycles_per_*`, but for larger values it think it would come out in the wash.

Reply:
http://www.agner.org/optimize/instruction_tables.pdf

Although RDTSC is one of the expensive ones, in my opinion, it is not comparable with packet processing. Default Bypass will not call it as well. The overall cycling measurement is anyhow including all cycles computing those.

Reply:
Ah, you're right! I ignored that `cur_tsc` gets set before this -.-